static/js/reducers/posts_for_channel.js

```diff
@@ -1,31 +1,28 @@
 // @flow
 import { GET, INITIAL_STATE } from 'redux-hammock/constants';
 
-import { makeChannel } from '../factories/channels';
-import { makeChannelPostList } from '../factories/posts';
+import * as api from '../lib/api';
 import type { Channel, Post } from '../flow/discussionTypes';
 
 type ChannelEndpointResponse = {
-  channel: Channel,
+  channelName: string,
   posts: Array<Post>
 };
 
 export const postsForChannelEndpoint = {
   name: 'postsForChannel',
   verbs: [ GET ],
-  getFunc: async (channelID: string) => {
-    let channel = makeChannel();
-    channel.name = channelID;
-    return {
-      channel: channel,
-      posts: makeChannelPostList()
-    };
+  getFunc: async (channelName: string) => {
+    return api.getPostsForChannel(channelName).then((posts) => ({
+      channelName,
+      posts,
+    }));
```

Comment:
a more `async` / `await` - ey wait to write this would be:

```js
getFunc: async (channelName: string) => {
  let posts = await api.getPostsForChannel(channelName);
  return { channelName, posts };
},
```

up to you if you want to write it that way or stick with what's here.